core/domain/collection_domain.py

```diff
@@ -1102,6 +1131,61 @@ def delete_skill(self, skill_id):
 
         del self.skills[skill_id]
 
+    def add_question_id_to_skill(self, skill_name, question_id):
+        """Adds the question id to the question list of the appropriate skill.
+
+        Args:
+            skill_name: str. The name of the skill.
+            question_id: str. The id of the question.
+
+        Raises:
+            Exception: question_id is already present in skill.
+        """
+        skill_id = None
+        for _, skill in self.skills.iteritems():
+            if skill.name == skill_name:
+                skill_id = skill.id
+        if skill_id is not None:
+            question_ids = self.skills[skill_id].question_ids
+            if question_id not in question_ids:
+                self.skills[skill_id].question_ids.append(
+                    question_id)
+            else:
+                raise Exception(
+                    '%s is already present in %s' % (
```

Comment:
Nit: perhaps 'Question ID %s is already present for skill %s', otherwise no one's going to know what the first arg means.

Reply:
Updated.